midi/midi/music_21.py

```diff
@@ -1,40 +1,45 @@
+import copy
 import numpy as np
 
+from typing import *
 from music21 import *
 try:
     from decode import get_sequence_of_notes, export_tempo_array
 except ImportError:
     from .decode import get_sequence_of_notes, export_tempo_array
 
 
-def check_number_of_tracks(array, filename, number):
+def check_number_of_tracks(array: list[list[Tuple[int, list[int]]]],
+                           number: int) -> None:
     """
     checks if number of tracks equals an expected value: if not, raises a ValueError
 
     :param array:
-    :param filename:
     :param number:
     :return:
     """
     if len(array) != number:
-        raise ValueError('number of tracks ({}) in {} not matching the expected value ({})'
-                         .format(len(array), filename, number))
+        raise Warning('number of tracks ({}) not matching the expected value ({})'
+                      .format(len(array), number))
 
 
-def get_event_lists(array):
+def get_event_lists(array: list[list[Tuple[int, list[int]]]]) -> Tuple[list[list[list[int]]], list[int]]:
     """
     translates a multi-track array representing notes into a list of events with active notes
     
     :param array: 
     :return: 
     """
-    events = [[] for _ in range(len(array) + 1)]
-    tuples = [None] * len(array)
+    array = copy.deepcopy(array)
+    events: list[list[list[int]]] = [[] for _ in range(len(array))]
+    tuples = dict[int, Tuple[int, list[int]]]()
+    # tuples: list[Tuple[int, list[int]]] = [] * len(array)
```

Comment:
debug?

Reply:
corrected in dc1a838446b74d0fcf07c89894ed4ef042469233